edn/src/edn.rustpeg

```diff
@@ -44,13 +44,31 @@ pub boolean -> Value =
     "false" { Value::Boolean(false) }
 
 digit = [0-9]
+alphanumeric = [0-9a-zA-Z]
+octaldigit = [0-7]
+hex = [0-9a-fA-F]
 sign = "-" / "+"
 
 pub bigint -> Value =
     b:$( sign? digit+ ) "N" {
         Value::BigInteger(b.parse::<BigInt>().unwrap())
     }
 
+pub octalinteger -> Value =
+    "0" i:$( octaldigit+ ) {
+        Value::Integer(i64::from_str_radix(i, 8).unwrap())
+    }
+
+pub hexinteger -> Value =
+    "0x" i:$( hex+ ) {
+        Value::Integer(i64::from_str_radix(i, 16).unwrap())
+    }
+
+pub basedinteger -> Value =
+    b:$(digit digit?) "r" i:$( alphanumeric+ ) {
```

Comment:
This throws when using a base that's not in `(2, 36)` -- but still gets parsed. Should this be handled differently?